code/go/0chain.net/smartcontract/errors.go

```diff
@@ -0,0 +1,136 @@
+	}
+	return fmt.Errorf("%w: %v", err, msg)
+}
+
+const noResourceErrMsg = "can't get resource"
+
+// noResourceErr represents error corresponds to http.StatusNotFound.
+type noResourceErr struct {
+	msg string
+}
+
+// Make sure noResourceErr implement error interface.
+var _ error = (*noResourceErr)(nil)
+
+func (e *noResourceErr) Error() string {
+	return e.msg
+}
+
+func WrapErrNoResource(err error) error {
+	return fmt.Errorf("%w: %s", NewErrNoResource(), err)
+}
+
+var noResource = &noResourceErr{msg: noResourceErrMsg}
+
+func NewErrNoResource() error {
+	return noResource
+}
+
+var (
+	FailRetrievingLimitsErr         = errors.New("can't get limits")
+	FailRetrievingConfigErr         = errors.New("can't get config")
+	FailRetrievingStatsErr          = errors.New("can't get stats")
+	MinerDoesntExistErr             = errors.New("unknown miner")
+	FailRetrievingUserNodeErr       = errors.New("can't get user node")
+	FailRetrievingMinerNodeErr      = errors.New("can't get miner node")
+	FailRetrievingMinersListErr     = errors.New("can't get miners list")
+	FailRetrievingShardersListErr   = errors.New("can't get sharders list")
+	FailRetrievingPhaseNodeErr      = errors.New("can't get phase node")
+	FailRetrievingMinersDKGListErr  = errors.New("can't get miners dkg list")
+	FailRetrievingMinersMpksListErr = errors.New("can't get miners mpks list")
+	FailRetrievingGroupErr          = errors.New("can't get group")
+	FailRetrievingMagicBlockErr     = errors.New("can't get magic block")
+	PoolStatsNotFoundErr            = errors.New("pool stats not found")
+	RetrievingGlobalNodeErr         = errors.New("can't get global node")
+	FailRetrievingReadMarker        = errors.New("can't get read marker")
+	FailRetrievingAllocationErr     = errors.New("can't retrieve allocation")
+	FailRetrievingAllocationList    = errors.New("can't get allocation list")
+	FailAllocationMinLockErr        = errors.New("allocation min lock failed")
+	NoRegisteredBlobberErr          = errors.New("no blobbers registered, failed to check min allocation lock")
+	NotEnoughBlobbersErr            = errors.New("not enough blobbers to honor the allocation")
+	FailRetrievingPreferredBlobbers = errors.New("can't get preferred blobbers")
+	BlobberChallengeReadErr         = errors.New("error reading blobber challenge from DB")
+	FailRetrievingBlobbersListErr   = errors.New("can't get blobbers list")
+	FailRetrievingBlobberErr        = errors.New("can't get blobber")
+	FailRetrievingReadPoolErr       = errors.New("can't get read pool")
+	FailRetrievingWritePoolErr      = errors.New("can't get write pool")
+	FailRetrievingStakePool         = errors.New("can't get related stake pool")
+	FailRetrievingUserStakePoolErr  = errors.New("can't get user stake pools")
+	FailRetrievingChallengePoolErr  = errors.New("can't get challenge pool")
+	FailRetrievingPoolErr           = errors.New("can't get pool")
+	FailGetOrCreateClientPoolsErr   = errors.New("can't get or create client pools")
+)
```

Comment:
Maybe we can define smartcontract specific errors inside corresponding smartcontract package?

Reply:
Sure, I've replaced it with string constants.